src/Properties/check_property.jl

```diff
@@ -145,10 +145,20 @@ function check_property(S::AbstractSystem,
     if lazy_inputs_interval == 0
         overapproximate_inputs_fun = (k, i, x) -> overapproximate_fun(i, x)
     else
-        overapproximate_inputs_fun =
-            (k, i, x) -> (k % lazy_inputs_interval == 0) ?
-                         overapproximate_fun(i, x) :
-                         x
+        # first set
+        function _f(k, i, x::LinearMap{MN, NUM}) where {MN, NUM}
+            return CacheMinkowskiSum(LazySet{NUM}[x])
+        end
+        # further sets
+        function _f(k, i, x::MinkowskiSum)
+            @assert (x.X isa CacheMinkowskiSum) "expected CacheMinkowskiSum type, got $(typeof(x.X))"
```

Comment:
why don't you dispatch on the type of the first argument? compare with

```julia
function overapproximate(S::ConvexHull{N, Zonotope{N}, Zonotope{N}},
                         ::Type{<:Zonotope})::Zonotope where {N<:Real}
...
end
```

we may also add one `function _f` that works even withut `x.X` being a `CacheMinkowskiSum`, if it could be used.

Reply:
This was on purpose. If the first argument is not a `CacheMinkowskiSum`, then something is broken resp. the idea of the lazy inputs does not work as I expected it.